source/architecture.rst

```diff
@@ -183,6 +183,56 @@ Container Diagram for Distribution Systems
 
 .. See commented code in S-ENDA-metadata-service-context-diagram.puml
 
+--------------------------------------
+S-ENDA Data Access Service C4 Diagrams
+--------------------------------------
+
+We categorize data consumers in three levels:
+
+* :ref:`advanced-consumers`
+* :ref:`intermediate-consumers`
+* :ref:`simple-consumers`
+
+
+S-ENDA advanced data access context
+===================================
+
+The system described here is concerned with :ref:`advanced-consumers` and :ref:`data-producers`.
+
+Functional requirements
+-----------------------
+
+* :ref:`data-producers` should be able to produce a dataset and upload results to the data access service without time-consuming transformations
+* :ref:`advanced-consumers` must be able to download a copy of the entire dataset
+* :ref:`advanced-consumers` must be able to stream and filter parts of the dataset
+* :ref:`advanced-consumers` need access to enough use metadata to be able to locally post-process, reproject, etc., the dataset
+* The data access service must support the FAIR principles, in particular (meta)data interoperability and reusability
+* The transport mechanism used in the data access service need to be a widely adopted standard solution, and it must be open-source
+* The dataset need to be on a widely adopted open data format standard
+* The data access service as a whole need to be easy to use with familiar tools from the meteorological/climate/oceanographic domain, both for upload and download
+* The data access service must work together with an event-driven production system
+* The data acccess service should support the Harmonised Data API from the European Weather Cloud
+* :ref:`advanced-consumers` need to able to give feedback on the data access service and each individual dataset.
+
+
+Qualitative requirements
```

Comment:
@mortenwh "Quality attributes" is the standard term used for non-functional requirements. Some people also use "non-functional requirements", but from the books I have read quality attributes are more used and a better term. I think its better to use the standardized terms for this instead of using our own variations.

Reply:
ok - yes, I agree!